lib/fetchers/local.rb

```diff
@@ -13,18 +13,23 @@ def self.resolve(target)
       local_path = if target.is_a?(String)
                      resolve_from_string(target)
                    elsif target.is_a?(Hash)
+                     # If target is a Hash then it is a dependency. We should
+                     # archive it in order to create a checksum and vendor it.
+                     do_vendor = true
```

Comment:
Is this doing a revendor of the local dependency if we execute the profile locally? We probably only want to (but lets discuss):
- do not vendor local profiles if we just execute them
- do vendor when we run inspec archive or inspec upload

Reply:
I'll have to look into it, I agree it should behave as you've laid out though.

Reply:
Looked into it. It looks like it will vendor to the user's `~/.inspec/cache` on `execute` same behavior as current.

I'm thinking about your request and I think we may want to have it not vendor on local profiles (for development purposes) and the change how `archive` and `upload` work to do the vendor.

Reply:
Nope...I lied, it currently does NOT vendor to the cache on `inspec exec` :face_with_head_bandage: 

Met with the team, we are going to:
  - Automatically vendor on `inspec archive`
  - Not vendor if using a local path and `inspec exec`

`inspec compliance upload` already does a vendor so no worry there.

Reply:
I think we need to vendor local directories if we use `inspec archive` or `inspec compliance upload`. Otherwise those profiles will never be able to run within Automate.